tests/pipe_test.cc

```diff
@@ -24,74 +24,138 @@ TEST_CASE("Pipe is checked", "[Pipe]") {
     const double head2 = 100.0;
 
-    // Dimensionless Darcy friction factor
+    // Pipe length
+    const double length = sqrt(3 * pow(1.1, 2));
+
+    // Gravity
+    const double gravity = 9.81;
+
+    // Dimensionless Darcy friction factor (for Darcy-Weisbach equation)
     const double darcy_friction = 0.1;
 
-    // Calculated discharge in pipe in m3/min using Darcy-Weisbach head loss
+    // Pipe roughness coefficient (for Hazen-Williams equation)
+    const double pipe_roughness = 100;
+
+    // Calculated discharge in pipe in m3/s using Darcy-Weisbach head loss
+    // equation
+    const double discharge_darcy_weisbach =
+        sqrt((head1 - head2) * pow(M_PI, 2) * gravity * pow(diameter, 5) /
+             (8. * darcy_friction * length));
+    // Calculated discharge in pipe in m3/s using Hazen-Williams head loss
     // equation
-    const double discharge =
-        sqrt(10 * pow(M_PI, 2) * 9.81 * pow(2 * 10, 5) / (8 * 0.1));
+    const double discharge_hazen_williams =
+        pow(((head1 - head2) * pow(pipe_roughness, 1.852) *
+             pow(diameter, 4.8704) / (10.67 * length)),
+            1 / 1.852);
 
     // Assign defined variables to nodes and pipe
     nodes.at(0)->head(head1);
     nodes.at(1)->head(head2);
-    pipe->radius(radius);
-    pipe->max_velocity(max_velocity);
-    pipe->darcy_friction(darcy_friction);
+    pipe2->darcy_friction(darcy_friction);
+    pipe2->pipe_roughness(pipe_roughness);
 
     // Check radius and max flow velocity of the pipe
-    REQUIRE(pipe->max_discharge() == Approx(max_discharge).epsilon(tolerance));
-    // Check discharge and Darcy friction factor of the pipe
-    REQUIRE(pipe->discharge() == Approx(discharge).epsilon(tolerance));
+    REQUIRE(pipe2->max_discharge() == Approx(max_discharge).epsilon(tolerance));
+
+    // Check discharge and Darcy friction factor of the pipe using
+    // Darcy-Weisbach equation
+    pipe2->compute_discharge_darcy_weisbach();
+    REQUIRE(pipe2->discharge() ==
+            Approx(discharge_darcy_weisbach).epsilon(tolerance));
+    // Calculated headloss in pipe in m using Darcy-Weisbach head loss equation
+    // from previous calculated discharge
+    const double headloss_darcy_weisbach =
+        (8. * length * darcy_friction * pow(discharge_darcy_weisbach, 2)) /
+        (pow(M_PI, 2) * gravity * pow(diameter, 5));
+    // Check headloss of the pipe using Darcy-Weisbach equation
+    pipe2->compute_headloss_darcy_weisbach();
+    REQUIRE(pipe2->headloss() ==
+            Approx(headloss_darcy_weisbach).epsilon(tolerance));
+
+    // Check discharge and pipe roughness coefficient of the pipe using
+    // Hazen-Williams equation
+    pipe2->compute_discharge_hazen_williams();
+    REQUIRE(pipe2->discharge() ==
+            Approx(discharge_hazen_williams).epsilon(tolerance));
+    // Calculated headloss in pipe in m using Hazen-Williams head loss equation
+    // from previous calculated discharge
+    const double headloss_hazen_williams =
```

Comment:
Can you calculate the actual value? manually by hand otherwise you are just copy-pasting the exact equation from the code above.

Reply:
Is it ok just adding hand calculation in the comment? 
Or should I preserve 10E-12 accuracy and put the hand calc result into the code? (pipe->discharge() == Approx(hand_calc_result).epsilon(10E-12))

Reply:
I'd add the hand_calc_result in the code, like you are saying